BuildResidentialHPXML/measure.rb

```diff
@@ -3388,7 +3388,7 @@ def self.create(runner, model, args, epw_path)
     end
 
     # Check for errors in the HPXML object
-    if args[:apply_validation].is_initialized && args[:apply_validation].get
+    if (args[:apply_defaults].is_initialized && args[:apply_defaults].get) || (args[:apply_validation].is_initialized && args[:apply_validation].get)
```

Comment:
That's good, but it doesn't cover the Schema/Schematron validation, which occurs [here](https://github.com/NREL/OpenStudio-HPXML/blob/apply-validation-defaults/BuildResidentialHPXML/measure.rb#L3085). And it occurs _after_ the HPXML defaulting, so somehow the order needs to be reversed.

Also, rather than trying to update lines of code where `apply_validation` is referenced, a better approach would be to just set `apply_validation` to true when `apply_defaults` is true. Then it will be guaranteed to apply all the appropriate validation _without the possibility of missing references_.

Reply:
Ah I see. So currently BuildResHPXML:
* builds up the `hpxml` object
* runs `hpxml.check_for_errors()` if `apply_validation=true`
* applies defaults if `apply_defaults=true`
* exports `hpxml` to file
* applies Schema/Schematron validation if `apply_validation=true`

We want to apply defaults only after BOTH (1) `hpxml.check_for_errors()` and (2) Schema/Schematron validation have occurred.

Reply:
If my previous comment is correct, wouldn't we need to then export `hpxml` to file twice? Once before defaulting, and then again after defaulting?

Reply:
I think that's right. But exporting to file is extremely fast -- I timed it and found it to be 0.02 seconds w/ WSL on my machine.

Reply:
To provide a bit more info... the `hpxml.check_for_errors()` call is additional error-checking that cannot be performed through Schematron, so we resort to checks in ruby. Realistically, all three validations (Schema, Schematron, ruby) should probably happen in the same place in the measure.